rules/windows/powershell/powershell_script/posh_ps_domainpasswordspray_attacks.yml

```diff
@@ -0,0 +1,33 @@
+title: DomainPasswordSpray Attacks
+id: 5ff65fa4-32a1-4773-90cc-6238bf784148
+status: test
+description: Detects performing a domain password spray using the DomainPasswordSpray tool.
+references:
+    - https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1110.003/T1110.003.md
+    - https://github.com/dafthack/DomainPasswordSpray
+author: Mustafa Kaan Demir (mkdemir)
+tags:
+    - attack.execution
+    - attack.t1110.003
+    - attack.t1059.001
+logsource:
+    product: windows
+    category: ps_script
+    definition: Script Block Logging must be enabled
+detection:
+    selection:
+        ScriptBlockText|contains:
+           - 'Invoke-DomainPasswordSpray'
+    selection_p1:
+        ScriptBlockText|contains:
+        - ' -Password'
+    selection_p2:
+        ScriptBlockText|contains:
+        - ' -UserList'
+        - ' -PasswordList'
+        - ' -Domain'
+        - ' -Force'
+    condition: selection and (selection_p1 or selection_p2)
+falsepositives:
+    - Unknow
+level: critical
```

Comment:
We don't usually create a specific rule for malicious cmdlets. What I suggest is to add this cmdlet to the following rules:

- Add the function name to this rule [89819aa4-bbd6-46bc-88ec-c7f7fe30efa6](https://github.com/SigmaHQ/sigma/blob/fca389bfef32acb00bc6dd56a9283cb9e1cb61e3/rules/windows/powershell/powershell_script/posh_ps_malicious_commandlets.yml). While looking at the file there is also an interesting function called `Invoke-SpraySinglePassword` please add it as well.
- Add the file name `DomainPasswordSpray.ps1` to the rule [f331aa1f-8c53-4fc3-b083-cc159bc971cb](https://github.com/SigmaHQ/sigma/blob/fca389bfef32acb00bc6dd56a9283cb9e1cb61e3/rules/windows/file/file_event/file_event_win_powershell_exploit_scripts.yml)

Reply:
Thank you for your feedback.

Reply:
Don't forget to add the peramlink of the github you mentioned as frack said.